apisix/core/request.lua

```diff
@@ -143,14 +143,35 @@ function _M.set_header(ctx, header_name, header_value)
     end
 end
 
-function _M.add_header(header_name, header_value)
+function _M.add_header(ctx, header_name, header_value)
```

Comment:
The function `add_header` and `set_header` are almost identical, maybe we could extract a common function

Reply:
The names follow the corresponding API in OpenResty.